src/compiler/tools/results/TextChange.ts

```diff
@@ -9,7 +9,6 @@ export class TextChange {
     /** @internal */
     private readonly _compilerObject: ts.TextChange;
 
-    /** @internal */
```

Comment:
Could you make this internal again? Also, add `/** @internal **/` to the constructors of all the other classes.

In the future, I'm thinking to change these to have private modifiers in the declaration file.

Reply:
Why don't you want to publish the constructor? Currently, SourceFile.applyChanges is very useful. When working with TypeScript plugins and I need to cheat the library to instantiate the tsa TextChange thing sourceFile.applyChanges( new (TextChange as any)(tsTextChanges) )...

Reply:
I often do breaking changes on the constructor.